src/__tests__/element-queries.js

```diff
@@ -1200,3 +1200,32 @@ it('gets form controls by label text on IE and other legacy browsers', () => {
   `)
   expect(getByLabelText('Label text').id).toBe('input-id')
 })
+
+// https://github.com/testing-library/dom-testing-library/issues/787
+it(`get the output element by it's label`, () => {
+  const {getByLabelText, rerender} = renderIntoDocument(`
+    <label>foo
+      <output>bar</output>
+    </label>
+  `)
+  expect(getByLabelText('foo')).toBeInTheDocument()
+
+  rerender(`
+    <label>
+      <small>foo</small>
+      <output>bar</output>
+    </label>
+  `)
+
+  expect(getByLabelText('foo')).toBeInTheDocument()
+})
+
+// https://github.com/testing-library/dom-testing-library/issues/343#issuecomment-555385756
+it(`should get element by it's label when there are elements with same text`, () => {
+  const {getByLabelText} = renderIntoDocument(`
+    <label>test 1
+      <textarea defaultValue="test"></textarea>
```

Comment:
I don't think "defaultValue" is a valid DOM attribute for `textarea` - it's a React concept
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea

Reply:
Yes you're right. I only added the end tag because it was not right. But I think that it should be fixed